File: trimming-middle/5000/data/collatz/src/collatz.js
// コラッツ数列のステップ数を計算する関数
function collatzSteps(n) {
  var steps = 0;
  while (n > 1) {
    if (n % 2 === 0) {
      n = n / 2;
    } else {
      n = 3 * n + 1;
    }
    steps++;
  }
  return steps;
}

// メイン処理
var targets = [27, 871, 6171];
console.log("Running Collatz Conjecture...");

for (var i = 0; i < targets.length; i++) {
  var n = targets[i];
  var steps = collatzSteps(n);
  console.log("Number: " + n + " -> Steps: " + steps);
}

File: trimming-middle/500/data/collatz/src/collatz.c
#include <stdio.h>

// コラッツ数列のステップ数を計算する関数
int collatzSteps(int n) {
    int steps = 0;
    while (n > 1) {
        if (n % 2 == 0) {
            n = n / 2;
        } else {
            n = 3 * n + 1;
        }
        steps++;
    }
    return steps;
}

int main() {
    // いくつかの数値でシミュレーションを実行
    int targets[] = {27, 871, 6171};
    int len = sizeof(targets) / sizeof(targets[0]);

    printf("Running Collatz Conjecture...\n");

    for (int i = 0; i < len; i++) {
        int n = targets[i];
        int steps = collatzSteps(n);
        printf("Number: %d -> Steps: %d\n", n, steps);
    }

    return 0;
}